modules/RTC/ortc/RTCPeerConnection.js

```diff
@@ -0,0 +1,532 @@
+                'iceTransport "icestatechange" event, '
+                + `state:${iceTransport.state}`);
+
+            this._emitIceConnectionStateChange();
+        };
+
+        // TODO: More stuff.
+
+        return iceTransport;
+    }
+
+    /**
+     */
+    _createOffer(options) { // eslint-disable-line no-unused-vars
+        if (this._closed) {
+            return Promise.reject(
+                new InvalidStateError('RTCPeerConnection closed'));
+        }
+
+        if (this.signalingState !== 'stable') {
+            return Promise.reject(new InvalidStateError(
+                `invalid signalingState "${this.signalingState}"`));
+        }
+
+        // TODO: More stuff.
+    }
+
+    /**
+     */
+    _createAnswer(options) { // eslint-disable-line no-unused-vars
+        if (this._closed) {
+            return Promise.reject(
+                new InvalidStateError('RTCPeerConnection closed'));
+        }
+
+        if (this.signalingState !== 'have-remote-offer') {
+            return Promise.reject(new InvalidStateError(
+                `invalid signalingState "${this.signalingState}"`));
+        }
+
+        // TODO: More stuff.
+    }
+
+    /**
+     */
+    _setLocalDescription(desc) {
+        if (this._closed) {
+            return Promise.reject(
+                new InvalidStateError('RTCPeerConnection closed'));
+        }
+
+        switch (desc.type) {
+        case 'offer':
+            if (this.signalingState !== 'stable') {
+                return Promise.reject(new InvalidStateError(
+                    `invalid signalingState "${this.signalingState}"`));
+            }
+
+            break;
+
+        case 'answer':
+            if (this.signalingState !== 'have-remote-offer') {
+                return Promise.reject(new InvalidStateError(
+                    `invalid signalingState "${this.signalingState}"`));
+            }
+
+            break;
+        }
+
+        // TODO: More stuff.
+    }
+
+    /**
+     */
+    _setRemoteDescription(desc) {
+        if (this._closed) {
+            return Promise.reject(
+                new InvalidStateError('RTCPeerConnection closed'));
+        }
+
+        switch (desc.type) {
+        case 'offer':
+            if (this.signalingState !== 'stable') {
+                return Promise.reject(new InvalidStateError(
+                    `invalid signalingState "${this.signalingState}"`));
+            }
+
+            break;
+
+        case 'answer':
+            if (this.signalingState !== 'have-local-offer') {
+                return Promise.reject(new InvalidStateError(
+                    `invalid signalingState "${this.signalingState}"`));
+            }
+
+            break;
+        }
+
+        // TODO: More stuff.
+    }
+
+    /**
+     */
+    _addStream(stream) {
+        if (this._closed) {
+            throw new InvalidStateError('RTCPeerConnection closed');
+        }
+
+        // TODO: Not sure. The app may wish to signal that the same stream
```

Comment:
In Chrome there is no need to call `addStream()` with an already attached stream. This is because Chrome's PC reacts on changes on the local MediaStream. That's deprecated however given the new "track based" API.

The point here is that I don't know *yet* whether jitsi calls twice `addStream()` with the same stream or not (I expect that does not happen).